users/vieting/experiments/switchboard/ctc/feat/baseline_args.py

```diff
@@ -237,6 +237,54 @@ def get_returnn_config(
         # network["source"] = specaug_layer_jingjing(in_layer=["features"])
         pass
 
+    if audio_perturbation and recognition:
+        feature_net = copy.deepcopy(feature_net)
+        audio_perturb_args = extra_args.get("audio_perturb_args", {})
+        source_layer = "data"
+        if "preemphasis" in audio_perturb_args:
+            for layer in feature_net["subnetwork"]:
+                if feature_net["subnetwork"][layer].get("from", None) == source_layer:
+                    feature_net["subnetwork"][layer]["from"] = "preemphasis"
+                    source_layer = "preemphasis"
```

Comment:
Perturbation is done first and preemphasis is done after in the network. If preemphasis is also done as a perturbation than it is done before aswell. I never did codec and preemphasis perturbation together, since both did not look promising. I am not sure what the correct order would be if I would do this

Reply:
I do not intend to do combination experiments with mu law and preemphasis atm. I could make them mutually exclusive, which would make the code simpler and avoid this problem.

Reply:
Yes, please do that. Otherwise, it can lead to hidden errors.